src/expr.rs

```diff
@@ -53,11 +55,37 @@ impl Rewrite for ast::Expr {
             ast::Expr_::ExprTup(ref items) => {
                 rewrite_tuple_lit(context, items, self.span, width, offset)
             }
+            ast::Expr_::ExprLoop(ref block, _) => {
+                // FIXME: this drops any comment between "loop" and the block.
+                block.rewrite(context, width, offset).map(|result| {
+                    format!("loop {}", result)
+                })
+            }
             _ => context.codemap.span_to_snippet(self.span).ok()
         }
     }
 }
 
+impl Rewrite for ast::Block {
+    fn rewrite(&self, context: &RewriteContext, _: usize, _: usize) -> Option<String> {
+        let mut visitor = FmtVisitor::from_codemap(context.codemap, context.config);
+        visitor.last_pos = self.span.lo;
+        visitor.block_indent = context.block_indent;
+
+        visitor.visit_block(self);
+
+        // Push text between last block item and end of block
+        let snippet = visitor.snippet(mk_sp(visitor.last_pos, self.span.hi));
+        visitor.changes.push_str_span(self.span, &snippet);
+
+        // Stringify visitor
+        let file_name = context.codemap.span_to_filename(self.span);
+        let string_buffer = visitor.changes.get(&file_name);
```

Comment:
I thought a bit about that when dealing with molules issue. In fact, there are only two places where we need the multiple-file aspect of the visitor: visiting a module and when we print the result. For all other tasks, complicated changeset is only noise. For example, there is no obvious reason for `ChangeSet::push_str_span` to need a span, the text should be added to the file we're working on automatically (we never make a `push_str_span` on more than one file).
A simple option to fix that would be to add a `currentFile: StringBuffer` to the visitor, all functions would add text on this, and `format_mod` would exchange this when needed.
Eventually, `format_mod` could become an entry point for the visit, and we could remove entierely the changeset of the visitor.